cmd/fabctx/eval_ctx.go

```diff
@@ -0,0 +1,91 @@
+package fabctx
+
+import (
+	"context"
+	"log/slog"
+	"maps"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/joho/godotenv"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+	"github.com/zclconf/go-cty/cty/function/stdlib"
+)
+
+// Creates a new eval context. Base eval context is cloned and extended with environment variables.
+func newEvalContext() *hcl.EvalContext {
+	vars := maps.Clone(baseEvalContext.Variables)
+	if vars == nil {
+		vars = make(map[string]cty.Value)
+	}
+	vars["env"] = buildEnvVarMap()
+	evalCtx := &hcl.EvalContext{
+		Variables: vars,
+		Functions: maps.Clone(baseEvalContext.Functions),
+	}
+	return evalCtx
+}
+
+func buildEnvVarMap() cty.Value {
+	envVars := os.Environ()
+	envFromFile, err := godotenv.Read()
+	if err != nil && !os.IsNotExist(err) {
+		slog.Error("Error reading .env file", "err", err)
+	}
+	envMap := make(map[string]cty.Value, len(envFromFile)+len(envVars))
+	for k, v := range envFromFile {
+		envMap[k] = cty.StringVal(v)
+	}
+	for _, e := range envVars {
```

Comment:
This will override dotenv variables with system variables. Usually `.env` variables should have higher priority.

Reply:
godotenv has a different opinion about that: 
https://github.com/joho/godotenv#precedence--conventions
https://github.com/joho/godotenv/blob/32e64fa834a9f5fcfe7506c2c603d6535da7eb83/godotenv.go#L184

I think that `.env` holds something more static, that shouldn't change from run to run (like API keys), and the env vars are something that could be modified directly before launching the `fabric` binary, and thus should take precedance over `.env` values.

Reply:
Hmm, actually makes sense. Having ability to override some vars before running CLI has usecases.